### PR TITLE
Add support for gasLimitOf function

### PR DESCRIPTION
**Pull request by adam-maj**

### CODE REVIEW

File: src/core/classes/gas-cost-estimator.ts [hunk 1]
```diff
@@ -39,6 +39,31 @@ export class GasCostEstimator<TContract extends BaseContract> {
...
+   * const gasLimitOfClaim = await nftDrop?.estimator.gasLimitOf("claim", [
+   *   "0x...", // receiver
+   *   1, // quantity
+   *   "0x...", // currency
+   *   1, // price per token
+   *   [], // proofs
+   *   1, // proof max quantity per transaction
+   * ]);
+   * ```
+   * @returns the estimated gas limit in GWEI
```
Comment (joaquim-verges) [inline src/core/classes/gas-cost-estimator.ts:57]: comment is wrong

File: src/core/classes/gas-cost-estimator.ts [hunk 2]
```diff
@@ -39,6 +39,31 @@ export class GasCostEstimator<TContract extends BaseContract> {
...
+   * @public
+   */
+  public async gasLimitOf(
+    fn: keyof TContract["functions"] | (string & {}),
+    args: Parameters<TContract["functions"][typeof fn]> | any[],
+  ): Promise<BigNumber> {
+    return this.contractWrapper.estimateGas(fn, args);
```
Comment (joaquim-verges) [inline src/core/classes/gas-cost-estimator.ts:64]: can you write a quick test for this for sanity? can just check that gasLimit > 0

File: src/core/classes/gas-cost-estimator.ts (hunk 1 above)
Comment (adam-maj) [reply src/core/classes/gas-cost-estimator.ts:57]: What about it were you referring to - something besides `wei` instead of `gwei`?

Comment (adam-maj) [reply src/core/classes/gas-cost-estimator.ts:57]: ah yes - just gas units

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**